platform/platform_filesystem.c

```diff
@@ -53,107 +54,195 @@ For more information, please refer to <http://unlicense.org>
 
 /*	File System	*/
 
-PLresult plInitFileSystem(void) {
-    return PL_RESULT_SUCCESS;
+/** FS Mounting **/
+
+typedef enum FSMountType {
+	FS_MOUNT_DIR,
+	FS_MOUNT_PACKAGE,
+} FSMountType;
+
+typedef struct PLFileSystemMount {
+	FSMountType type;
+	union {
+		PLPackage* pkg;                        /* FS_MOUNT_PACKAGE */
+		char path[PL_SYSTEM_MAX_PATH];   /* FS_MOUNT_DIR */
+	};
+	struct PLFileSystemMount* next, * prev;
+} PLFileSystemMount;
+static PLFileSystemMount* fs_mount_root = NULL;
+static PLFileSystemMount* fs_mount_ceiling = NULL;
+
+void plClearMountedLocation( PLFileSystemMount* location ) {
+	if ( location->type == FS_MOUNT_PACKAGE ) {
+		plDestroyPackage( location->pkg );
+		location->pkg = NULL;
+	}
+
+	location->prev->next = location->next;
+	location->next->prev = location->prev;
+
+	/* ensure root and ceiling are always pointing to a valid location */
+	if ( location == fs_mount_root ) {
+		fs_mount_root = location->next;
+	}
+	if ( location == fs_mount_ceiling ) {
+		fs_mount_ceiling = location->prev;
+	}
+
+	pl_free( location );
+}
+
+/**
+ * Clear all of the mounted locations
+ */
+void plClearMountedLocations( void ) {
+	while ( fs_mount_root != NULL ) { plClearMountedLocation( fs_mount_root ); }
 }
 
-void plShutdownFileSystem(void) {}
+static void _plInsertMountLocation( PLFileSystemMount* location ) {
+	if ( fs_mount_root == NULL ) {
+		fs_mount_root = location;
+	}
+
+	location->prev = fs_mount_ceiling;
+	fs_mount_ceiling->next = location;
```

Comment:
Should check `fs_mount_ceiling` isn't NULL before assigning into it. Also initialise `location->next` to NULL